velox/exec/GroupingSet.cpp

```diff
@@ -103,9 +99,34 @@ GroupingSet::GroupingSet(
         allAreSinglyReferenced(aggregate.inputs, channelUseCount));
   }
 
+  if (!aggregates_.empty()) {
+    bool hasPartialInput{false};
+    bool hasPartialOutput{false};
+    bool allPartialInput{true};
+    bool allPartialOutput{true};
+
+    for (const auto& aggregate : aggregates_) {
+      if (core::AggregationNode::Aggregate::isPartialInput(aggregate.step)) {
+        hasPartialInput = true;
+      } else {
+        allPartialInput = false;
+      }
+      if (core::AggregationNode::Aggregate::isPartialOutput(aggregate.step)) {
+        hasPartialOutput = true;
+      } else {
+        allPartialOutput = false;
+      }
+    }
+
+    hasPartialInput_ = hasPartialInput;
+    hasPartialOutput_ = hasPartialOutput;
+    allPartialInput_ = allPartialInput;
+    allPartialOutput_ = allPartialOutput;
+  }
+
   sortedAggregations_ =
       SortedAggregations::create(aggregates_, inputType, &pool_);
-  if (isPartial_) {
+  if (hasPartialOutput_) {
```

Comment:
This check doesn't seem correct. I assume it should be fine if none of sortedAggregations_ is partial, no?

Reply:
> I assume it should be fine if none of sortedAggregations_ is partial

This is true because the check will pass if non of sortedAggregations_ is partial as `hasPartialOutput_` = `false`.

But would you intentionally mean "it should be fine if SOME of sortedAggregations_ are partial"? Which will suggest the change

```
  if (allPartialOutput_) {
    VELOX_USER_CHECK_NULL(
        sortedAggregations_,
        "Partial aggregations over sorted inputs are not supported");
  }
```

? I think it makes sense but can be done in the future because in the first PR I am only changing the APIs. So I put the strongest restriction here.